beater/server.go

```diff
@@ -90,16 +87,23 @@ func stop(server *http.Server, timeout time.Duration) {
 	}
 }
 
-type handler func(w http.ResponseWriter, r *http.Request)
-
-func createHandler(p processor.Processor, config Config, report reporter) handler {
-	return func(w http.ResponseWriter, r *http.Request) {
-		logp.Debug("handler", "Request: URI=%s, method=%s, content-length=%d", r.RequestURI, r.Method, r.ContentLength)
+func authHandler(secretToken string, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		requestCounter.Inc()
+		if !isAuthorized(r, secretToken) {
+			sendStatus(w, r, 401, errInvalidToken)
+			return
+		}
+		h.ServeHTTP(w, r)
+	})
+}
 
+func appHandler(p processor.Processor, config Config, report reporter) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logp.Debug("handler", "Request: URI=%s, method=%s, content-length=%d", r.RequestURI, r.Method, r.ContentLength)
```

Comment:
With this change a request is is not logged any more when it is not authorized.

Reply:
yes,I was looking for a better way to log.Could not use reportInfo in sendStatus as I wanted to.